tests/testthat/test-joins.R

```diff
@@ -92,3 +92,8 @@ test_that("sql generated correctly for all sources", {
 
   expect_equal_tbls(xy)
 })
+
+test_that("full join is promoted to cross join for no overlapping variables", {
+  result <- df1 %>% full_join(df2) %>% collect()
```

Comment:
I don't like making this the default because if you accidentally supply the wrong data frames you might create a query that takes hours to run. Could we make it explicit, i.e. `by = character()`?

Reply:
Changed.